Tangem/Modules/Onboarding/OnboardingStepsBuilders/WalletOnboardingStepsBuilder.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  WalletOnboardingStepsBuilder.swift
+//  Tangem
+//
+//  Created by Alexander Osokin on 06.04.2023.
+//  Copyright © 2023 Tangem AG. All rights reserved.
+//
+
+import Foundation
+import TangemSdk
+
+class WalletOnboardingStepsBuilder {
+    private let card: CardDTO
+    private let backupService: BackupService
+    private let touId: String
+
+    private var userWalletSavingSteps: [WalletOnboardingStep] {
+        guard BiometricsUtil.isAvailable,
+              !AppSettings.shared.saveUserWallets,
+              !AppSettings.shared.askedToSaveUserWallets else {
+            return []
+        }
+
+        return [.saveUserWallet]
+    }
+
+    private var backupSteps: [WalletOnboardingStep] {
+        if card.backupStatus?.isActive == true {
+            return []
+        }
+
+        if !card.settings.isBackupAllowed {
+            return []
+        }
+
+        var steps: [WalletOnboardingStep] = .init()
```

Comment:
NP: просто глаз зацепился ) 

```suggestion
        var steps: [WalletOnboardingStep] = []
```